### DISCUSSION TITLE
compose:args changes the position

### DISCUSSION THREAD

**Discussion by DOEHOONLEE** [ImageMagick/ImageMagick#4302] [Help]
![displace](https://user-images.githubusercontent.com/20305442/135613772-1dc9ef84-7bc1-426b-93b7-3a9bde16e2d3.png)

![andie](https://user-images.githubusercontent.com/20305442/135613778-c26e1ca3-a5e5-472f-a116-7e8d19779328.png)

With these images, by running the following command,

`convert andie.png displace.png -alpha set -virtual-pixel transparent -compose displace -set option:compose:ar
gs -15x-15 -composite imageWrinkled.png`

I do get an image with wrinkles like this

![imageWrinkled](https://user-images.githubusercontent.com/20305442/135614425-9e83fe2b-2262-42d9-9960-25cb64b85b42.png)

It is very realistic. However, the thing is that the image off by the value I put after `compose:args`

Image get its wrinkles nice and lovely with the arguments put in, but it changes the position of the overlay pattern image that I put on the shirt.

It does not get wrinkles without this `compose:args`

Is there any solution to this problem? Should I apply the wrinkles then move the image by the value that I applied to the `displace compose:args`??

Thank you always!! and Happy Coding!!

**Comment by snibgo** [comment]
What version of IM, on what platform?

I don't understand what problem you want to solve.

You have a photo of a sheet, and are using that to displace an image of a droid.

Where the sheet is 50% gray, there is no displacement. Where the sheet is darker, the droid is displaced up and left. Where the sheet is lighter, the droid is displaced down and right.

Most of the droid is displaced down and right because most of the sheet is lighter than 50%. If you don't want this, then you can darken the sheet, eg with "-auto-gamma".
```
magick droid.png ( sheet2.png -auto-gamma ) -alpha set -virtual-pixel transparent -compose displace -set option:compose:args -15x-15 -composite out.png
```

**Comment by fmw42** [comment]
Shifting can be mitigated by ensuring the average gray level of the wrinkles image is mid-gray (gray=0.5 out of 1). So you can do that as follows. The first and last commands are optional just to show the average gray level before and during processing.

Input:

![droid](https://user-images.githubusercontent.com/13458218/136258876-0e6ac798-0ba1-4746-a677-9575f65b06b1.png)

Wrinkle image:

![wrinkles](https://user-images.githubusercontent.com/13458218/136258905-c8089e31-8dd1-4e47-a54d-94fdd9795bb0.png)

Wrinkle image graylevel before:

`convert wrinkles.png -format "%[fx:mean]" info:`
0.678061

Processing:

```
infile="droid.png"
grayfile="wrinkles.png"
diff=`convert "$grayfile" -format "%[fx:100*(0.5-mean)]" info:`
convert "$infile" \
\( "$grayfile" -evaluate add ${diff}% +write x.png \) \
-virtual-pixel transparent -compose displace \
-set option:compose:args -15x-15 -composite droid_wrinkled.png

```

Wrinkle gray level image during processing (See temporary file x.png)

`convert x.png -format "%[fx:mean]" info:`
0.49809

Result:

![droid_wrinkled](https://user-images.githubusercontent.com/13458218/136259113-318158db-9fb0-4b25-b173-1ae01e964e15.png)
